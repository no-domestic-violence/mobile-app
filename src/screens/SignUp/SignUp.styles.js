import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  view: {
    justifyContent: 'flex-end',
    height: '100%',
  },
  text: {
    fontSize: 14,
    color: '#000',
    marginTop: 20,
    fontStyle: 'italic',
  },
  actionsContainer: {
    marginBottom: 60,
  },
});
