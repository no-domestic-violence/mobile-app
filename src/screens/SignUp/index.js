import React, {useContext, useState} from 'react';
import {View, Text, Button, TextInput, StyleSheet, TouchableOpacity} from 'react-native';
// import {AuthContext} from '../../context';
import {signup} from '../../api/mock';

const styles = StyleSheet.create({
  input: {
    width: 350,
    height: 55,
    backgroundColor: '#fff',
    margin: 10,
    padding: 8,
    color: '#000',
    borderRadius: 14,
    fontSize: 18,
    fontWeight: '500',
  },
  view: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  button: {
    backgroundColor: '#009688',
    paddingLeft: 30,
    paddingRight: 30,
    paddingTop: 5,
    paddingBottom: 5,
    borderColor: 'white',
    borderWidth: 1,
    borderRadius: 12,
    color: 'white',
    fontSize: 20,
    overflow: 'hidden',
    textAlign:'center',
  }
});

export default function SignUpScreen({navigation}) {
  // const {signUp} = useContext(AuthContext);
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [loading, setLoading] = useState(false);
  const [errorMessage, setErrorMessage] = useState('');

  const handleUsername = (e) => {
    console.log(e.target.value);
  };
  const handlePassword = (e) => {
    console.log(e.target.value);
  };
  const handleEmail = (e) => {
    console.log(e.target.value);
  };

  const signUpUser = () => {
    setErrorMessage('');
    signup('test@test.com', 'password')
      .then(() => {
        navigation.navigate('Home');
      })
      .catch((e) => setErrorMessage(e.message));
  };

  return (
    <View  style={styles.view}>
      <TextInput
        style={styles.input}
        placeholder="Username"
        autoCapitalize="none"
        placeholderTextColor="#6c757d"
        onChange={handleUsername}
      />
      <TextInput
        style={styles.input}
        placeholder="Password"
        secureTextEntry={true}
        autoCapitalize="none"
        placeholderTextColor="#6c757d"
        onChangeText={handlePassword}
      />
      <TextInput
        style={styles.input}
        placeholder="Email"
        autoCapitalize="none"
        placeholderTextColor="#6c757d"
        onChangeText={handleEmail}
      />
      <TouchableOpacity onPress={() => signUpUser()}>
          <Text style={styles.button}>Sign Up</Text>
        </TouchableOpacity>
      {errorMessage ? <Text>{errorMessage}</Text> : null}
    </View>
  );
}


