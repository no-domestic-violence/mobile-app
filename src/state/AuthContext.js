import AsyncStorage from '@react-native-async-storage/async-storage';
import createAppContext from './CreateAppContext';
import appApiClient from '../api/appApiClient';

const authReducer = (state, action) => {
  switch (action.type) {
    case 'SIGNUP_ERROR':
    case 'LOGIN_ERROR':
      return {
        ...state,
        errorMessage: action.payload,
      };
    case 'SIGNUP_SUCCESS':
    case 'LOGIN_SUCCESS':
      return {
        ...state,
        token: action.payload.token,
        username: action.payload.user.username,
        errorMessage: '',
      };
    case 'CHANGE_PASSWORD_SUCCESS':
      return {
        ...state,
        successMessage: action.payload,
      };
    case 'CHANGE_PASSWORD_ERROR':
      return {
        ...state,
        errorMessage: action.payload,
      };
    case 'AUTH_SUCCESS':
      return {
        ...state,
        token: action.payload.token,
        username: action.payload.username,
      };
    case 'LOGOUT':
      return {
        token: '',
        username: '',
      };
    case 'DELETE_ACCOUNT':
      return {
        token: '',
        username: '',
        successMessage: action.payload.data,
      };
    case 'REMOVE_ERRORS':
      return {
        ...state,
        errorMessage: '',
      };
    case 'REMOVE_MESSAGES':
      return {
        ...state,
        successMessage: '',
      };
    default:
      return state;
  }
};

const signup = (dispatch) => async ({ email, password, username }) => {
  try {
    const response = await appApiClient.post('/signup', {
      email,
      password,
      username,
    });

    await AsyncStorage.setItem('token', response.data.token);
    await AsyncStorage.setItem('username', response.data.user.username);

    dispatch({ type: 'SIGNUP_SUCCESS', payload: response.data });
  } catch (error) {
    dispatch({
      type: 'SIGNUP_ERROR',
      payload: "Something went wrong'( Try again",
    });
  }
};

const login = (dispatch) => async ({ email, password }) => {
  try {
    const response = await appApiClient.post('/login', { email, password });
    await AsyncStorage.setItem('token', response.data.token);
    await AsyncStorage.setItem('username', response.data.user.username);
    dispatch({ type: 'LOGIN_SUCCESS', payload: response.data });
  } catch (error) {
    dispatch({
      type: 'LOGIN_ERROR',
      payload: 'Are you sure about password and email?',
    });
  }
};

const authentication = (dispatch) => async () => {
  const token = await AsyncStorage.getItem('token');
  const username = await AsyncStorage.getItem('username');
  const user = { token, username };
  if (token) {
    dispatch({ type: 'AUTH_SUCCESS', payload: user });
  }
};
const removeErrors = (dispatch) => () => {
  dispatch({ type: 'REMOVE_ERRORS' });
};

const removeMessages = (dispatch) => () => {
  dispatch({ type: 'REMOVE_MESSAGES' });
};

const changePassword = (dispatch) => async ({
  email,
  password,
  oldPassword,
}) => {
  try {
    const response = await appApiClient.post('/changePassword', {
      email,
      oldPassword,
      password,
    });
    removeErrors();
    dispatch({ type: 'CHANGE_PASSWORD_SUCCESS', payload: response.data });
  } catch (error) {
    dispatch({
      type: 'CHANGE_PASSWORD_ERROR',
      payload: error.response.data.message,
    });
  }
};

const signout = (dispatch) => async () => {
  await AsyncStorage.removeItem('token');
  dispatch({ type: 'LOGOUT' });
};

const deleteAccount = (dispatch) => async ({ username }) => {
  try {
    // TODO: fix sending params so it is more readable
    const response = await appApiClient.delete('/deleteUser', {
      params: { username },
    });
    await AsyncStorage.removeItem('token');
    await AsyncStorage.removeItem('username');
    dispatch({ type: 'DELETE_ACCOUNT', payload: response.data });
  } catch (error) {
    dispatch({
      type: 'CHANGE_PASSWORD_ERROR',
      payload: error.response.data.message,
    });
  }
};

export const { Provider, Context } = createAppContext(
  authReducer,
  {
    signup,
    login,
    signout,
    removeErrors,
    removeMessages,
    authentication,
    changePassword,
    deleteAccount,
  },
  { isLoggedIn: false, errorMessage: '' }
);
