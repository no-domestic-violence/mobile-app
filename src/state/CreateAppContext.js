import React, { useReducer } from 'react';

// boilerplate for creating Context and Provider
// does not to be changed
export default (reducer, actions, defaultValue) => {
  const Context = React.createContext();

  const Provider = ({ children }) => {
    const [state, dispatch] = useReducer(reducer, defaultValue);

    //actions is object for ex {updateCurrentLocation: (dispatch)=> (location) => { dispatch({ type: 'UPDATE_LOCATION', payload: location });}
    const boundActions = {};
    //we need to loop through each action in object and for every key(which is function) we call it with dispatch argument and it gives back function. this function we pass down to all children like ...boundActions
    for (const key in actions) {
      boundActions[key] = actions[key](dispatch);
    }
    return (
      <Context.Provider value={{ state, ...boundActions }}>
        {children}
      </Context.Provider>
    );
  };
  return { Context, Provider };
};
