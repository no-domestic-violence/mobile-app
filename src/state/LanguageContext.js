import React, { createContext, useState } from 'react';
import AsyncStorage from '@react-native-async-storage/async-storage';
import i18n from 'services/i18n';
import PropTypes from 'prop-types';
import { DEFAULT_LANGUAGE } from '../services/i18n/language-detector';

const APP_LANGUAGE = 'lang';

const availableLanguages = [
  { lang: 'en', name: 'English' },
  { lang: 'de', name: 'German' },
];

export const LanguageContext = createContext({
  availableLanguages,
  appLanguage: DEFAULT_LANGUAGE,
  setAppLanguage: () => {},
  initializeAppLanguage: () => {},
});

export const LanguageProvider = ({ children }) => {
  const [appLanguage, setAppLanguage] = useState(DEFAULT_LANGUAGE);

  const setLanguage = async (language) => {
    i18n.changeLanguage(language);
    setAppLanguage(language);
    AsyncStorage.setItem(APP_LANGUAGE, language);
  };

  const initializeAppLanguage = async () => {
    const currentLanguage = await AsyncStorage.getItem(APP_LANGUAGE);
    if (!currentLanguage) {
      setLanguage(DEFAULT_LANGUAGE);
    } else {
      setLanguage(currentLanguage);
    }
  };

  return (
    <LanguageContext.Provider
      value={{
        availableLanguages,
        setAppLanguage: setLanguage,
        appLanguage,
        initializeAppLanguage,
      }}>
      {children}
    </LanguageContext.Provider>
  );
};
LanguageProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
