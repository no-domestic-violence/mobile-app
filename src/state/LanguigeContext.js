//TODO: it did not work, to figure out if we need or we can just you provider for everything


// import React, { createContext } from 'react';
// import { I18nextProvider } from 'react-i18next';
// import i18next from 'i18next';

// const LanguigeContext = createContext();

// function LanguigeProvider(props) {

//   return (
//     <LanguigeContext.Provider>
//       <I18nextProvider i18n={i18next}>{props.children}</I18nextProvider>
//    </LanguigeContext.Provider>
//   );
// }

// export {LanguigeProvider, LanguigeContext};

