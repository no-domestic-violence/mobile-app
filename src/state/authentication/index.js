import createAppContext from 'helpers/createAppContext';
import {
  signup,
  login,
  signout,
  removeErrors,
  removeMessages,
  authentication,
  changePassword,
  deleteAccount,
  setAlreadyLaunchedValue,
} from './authentication.actions';
import authReducer from './authentication.reducer';

export const { Provider, Context } = createAppContext(
  authReducer,
  {
    signup,
    login,
    signout,
    removeErrors,
    removeMessages,
    authentication,
    changePassword,
    deleteAccount,
    setAlreadyLaunchedValue,
  },
  { isLoggedIn: false, errorMessage: '', isFirstLaunch: null }
);
