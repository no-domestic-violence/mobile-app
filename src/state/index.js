import React, {createContext, useState, useMemo} from 'react';

const AppContext = createContext();

function ContextProvider(props) {
  //just an example of context usage
  const [userToken, setUserToken] = useState('');
  const authContext = useMemo(() => {
    return {
      logIn: () => {
        setUserToken('tokenishere');
      },
      signUp: () => {
        setUserToken('tokenishere');
      },
      signOut: () => {
        setUserToken('');
      },
    };
  });

  return (
    <AppContext.Provider value={{authContext, userToken}}>
      {props.children}
    </AppContext.Provider>
  );
}

export {ContextProvider, AppContext};
