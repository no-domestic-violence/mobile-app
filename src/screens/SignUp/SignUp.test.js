import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import { AuthContext, AuthProvider } from 'state/index';
import SignUp from 'screens/SignUp/index';
import * as SecureStore from 'expo-secure-store';

const email = 'test@test.com';
const password = '12345678';
const username = 'celeste';
const token =
  'eyJ0eXAiOiJKV1QiLCJub25jZSI6IjJQSnRPNkMxR2hkaHB2VFl3ZVVseEpkMU5WbzJVZE0waUlwOHFXdHVMejgiLCJhbGciOiJSUzI1NiIsIng1dCI6ImtnMkxZczJUMENUaklmajRydDZKSXluZW4zOCIsImtpZCI6ImtnMkxZczJUMENUaklmajRydDZKSXluZW4zOCJ9.eyJhdWQiOiIwMDAwMDAwMy0wMDAwLTAwMDAtYzAwMC0wMDAwMDAwMDAwMDAiLCJpc3MiOiJodHRwczovL3N0cy53aW5kb3dzLm5ldC8wMTY1MWZjYS0zMzRjLTQxY2EtOWViYi04Y2FmNjkyMTIwZGEvIiwiaWF0IjoxNjA2MzM5NTg5LCJuYmYiOjE2MDYzMzk1ODksImV4cCI6MTYwNjM0MzQ4OSwiYWNjdCI6MCwiYWNyIjoiMSIsImFjcnMiOlsidXJuOnVzZXI6cmVnaXN0ZXJzZWN1cml0eWluZm8iLCJ1cm46bWljcm9zb2Z0OnJlcTEiLCJ1cm46bWljcm9zb2Z0OnJlcTIiLCJ1cm46bWljcm9zb2Z0OnJlcTMiLCJjMSIsImMyIiwiYzMiLCJjNCIsImM1IiwiYzYiLCJjNyIsImM4IiwiYzkiLCJjMTAiLCJjMTEiLCJjMTIiLCJjMTMiLCJjMTQiLCJjMTUiLCJjMTYiLCJjMTciLCJjMTgiLCJjMTkiLCJjMjAiLCJjMjEiLCJjMjIiLCJjMjMiLCJjMjQiLCJjMjUiXSwiYWlvIjoiRTJSZ1lLanZlaG4rVVBXMXBaLzhTZ1hwUy8xR0pUclQvSGovQ214Ny9PYTU4RFhPMDJjQiIsImFtciI6WyJwd2QiXSwiYXBwX2Rpc3BsYXluYW1lIjoiQm9uaWFsIENvbnNvbGUiLCJhcHBpZCI6ImY5NTk2MzU1LTQ0YmItNDAyZS04MWY3LTliZGY1YTJjYmNkNiIsImFwcGlkYWNyIjoiMCIsImZhbWlseV9uYW1lIjoiR29uemFsZXoiLCJnaXZlbl9uYW1lIjoiU2VyZ2lvIiwiaWR0eXAiOiJ1c2VyIiwiaXBhZGRyIjoiNjIuOTYuNzguMTE0IiwibmFtZSI6IlNlcmdpbyBHb256YWxleiIsIm9pZCI6IjQ5ZjZiMjdiLWM4YjYtNDM0NC1hMzY2LWQwOTA4YzcwNzQ4YSIsIm9ucHJlbV9zaWQiOiJTLTEtNS0yMS0zNTUyNjE5OTcwLTMzODgzOTQ4MjQtMjM1Mzk4MDkwNi0xMTE5IiwicGxhdGYiOiI1IiwicHVpZCI6IjEwMDMzRkZGOTg3QkJFMUYiLCJyaCI6IjAuQUFBQXloOWxBVXd6eWtHZXU0eXZhU0VnMmxWaldmbTdSQzVBZ2ZlYjMxb3N2TlpmQVBJLiIsInNjcCI6IkRpcmVjdG9yeS5BY2Nlc3NBc1VzZXIuQWxsIERpcmVjdG9yeS5SZWFkLkFsbCBvcGVuaWQgcHJvZmlsZSBVc2VyLlJlYWQgZW1haWwiLCJzaWduaW5fc3RhdGUiOlsia21zaSJdLCJzdWIiOiJxNjAzS0R1Z28wenlqa1lSeU5kUEpTa2NBT2gybEJ3TFA5V1B1azVZSVpBIiwidGVuYW50X3JlZ2lvbl9zY29wZSI6IkVVIiwidGlkIjoiMDE2NTFmY2EtMzM0Yy00MWNhLTllYmItOGNhZjY5MjEyMGRhIiwidW5pcXVlX25hbWUiOiJzZXJnaW8uZ29uemFsZXpAYm9uaWFsLmNvbSIsInVwbiI6InNlcmdpby5nb256YWxlekBib25pYWwuY29tIiwidXRpIjoiZ2hVOUVISF9nMHVRR3NQRTdDdlRBQSIsInZlciI6IjEuMCIsInhtc19zdCI6eyJzdWIiOiJkUnJyYXhyMk9INVEyc1NQUTdVeFRDcUUxeXUzNkVOVHJYTUVqMk1ZLWRvIn0sInhtc190Y2R0IjoxNDU0NjIwMTI0fQ.ZT89tyBnQ-oe8W3xCX7VYwSTV2pjyYzkFNtDVgwvD06h12-v-1QWI9FkAgwwGfW4kJZK4KqgA30LrpbX0BIOhxFuS1SaUrDnW-xgJ6kHBZZym9cyncWBQsVBphVB-850lkiHSLZvbgkriv8diSuEdjAhdewk7EFKYxyr0AG9boSjnpzXi2DAYLG_EVGwxfDyZ9DsdxSug5bhLar3avRFqBrnC1MO7vqBRdNirguzK4GWnrG5a48N_eVmeN-F46DzRUs1Q8QSlYD8ajsq9QjQ3osJdz2wgCF0bRJ0q8_Dyae0Pw8bqppEXlb1ls0o8blkLSvkcGRc8DPKIRqqTzAi6Q';

const navigate = jest.fn();
const addListener = jest.fn();
describe('Sign Up screen', () => {
  it('should match snapshot', () => {
    const result = render(
      <AuthProvider>
        <SignUp navigation={{ navigate, addListener }} />
      </AuthProvider>
    ).toJSON();
    expect(result).toMatchSnapshot();
  });
  it('should render default screen elements', () => {
    const { getAllByText, getByPlaceholderText, queryByText } = render(
      <AuthProvider>
        <SignUp navigation={{ navigate, addListener }} />
      </AuthProvider>
    );
    expect(getAllByText(/sign up/i).length).toBe(2);
    expect(getByPlaceholderText(/username/i));
    expect(getByPlaceholderText(/email/i));
    expect(getByPlaceholderText(/password/i));
    expect(queryByText(/Have an account? Go to login/i));
  });

  it('should show error messages on empty inputs submit', async () => {
    // given
    const { getByTestId, getByText } = render(
      <AuthProvider>
        <SignUp navigation={{ navigate, addListener }} />
      </AuthProvider>
    );
    // when
    fireEvent.press(getByTestId('signUp'));
    // then
    await waitFor(() => expect(getByText(/please enter your username/i)));
    await waitFor(() => expect(getByText(/please enter an email/i)));
    await waitFor(() =>
      expect(getByText(/please enter 8 characters password/i))
    );
  });
  it('should not show error message on valid username', async () => {
    // given
    const { getByTestId, queryByText, getByPlaceholderText } = render(
      <AuthProvider>
        <SignUp navigation={{ navigate, addListener }} />
      </AuthProvider>
    );
    // when
    fireEvent.changeText(getByPlaceholderText(/username/i), username);
    fireEvent.press(getByTestId('signUp'));
    // then
    await waitFor(() =>
      expect(queryByText(/please enter your username/i)).toBeNull()
    );
  });
  it('should show error on invalid password', async () => {
    // given
    const invalidPassword = 'one';
    const { getByTestId, getByText, getByPlaceholderText } = render(
      <AuthProvider>
        <SignUp navigation={{ navigate, addListener }} />
      </AuthProvider>
    );
    // when
    fireEvent.changeText(getByPlaceholderText(/password/i), invalidPassword);
    fireEvent.press(getByTestId('signUp'));
    // then
    await waitFor(() =>
      expect(getByText(/please enter 8 characters password/i))
    );
  });
  it('should not show error on valid password', async () => {
    // given
    const { getByTestId, queryByText, getByPlaceholderText } = render(
      <AuthProvider>
        <SignUp navigation={{ navigate, addListener }} />
      </AuthProvider>
    );
    // when
    fireEvent.changeText(getByPlaceholderText(/password/i), password);
    fireEvent.press(getByTestId('signUp'));
    // then
    await waitFor(() =>
      expect(queryByText(/please enter 8 characters password/i)).toBeNull()
    );
  });

  it('should show error on invalid email', async () => {
    // given
    const invalidEmail = 'test.com';
    const { getByTestId, getByText, queryByPlaceholderText } = render(
      <AuthProvider>
        <SignUp navigation={{ navigate, addListener }} />
      </AuthProvider>
    );
    // when
    fireEvent.changeText(queryByPlaceholderText(/email/i), invalidEmail);
    fireEvent.press(getByTestId('signUp'));
    // then
    await waitFor(() => expect(getByText(/email must be a valid email/i)));
  });
  it('should not show error on valid email', async () => {
    // given
    const { getByTestId, queryByText, getByPlaceholderText } = render(
      <AuthProvider>
        <SignUp navigation={{ navigate, addListener }} />
      </AuthProvider>
    );
    // when
    fireEvent.changeText(getByPlaceholderText(/email/i), email);
    fireEvent.press(getByTestId('signUp'));
    // then
    await waitFor(() =>
      expect(queryByText(/email must be a valid email/i)).toBeNull()
    );
  });
  it('should navigate to login screen on press "go to login"', async () => {
    // given
    const { getByText } = render(
      <AuthProvider>
        <SignUp navigation={{ navigate, addListener }} />
      </AuthProvider>
    );
    // when
    fireEvent.press(getByText(/go to login/i));
    // then
    await expect(navigate).toHaveBeenCalledWith('Login');
  });
  it('should handle valid input onSubmit and call signup action creator function with email, password and  username', async () => {
    // given
    const state = {
      isLoggedIn: true,
      errorMessage: '',
      isFirstLaunch: false,
      token,
    };
    await SecureStore.setItemAsync('token', JSON.stringify(token));
    await SecureStore.setItemAsync('username', JSON.stringify(username));
    const signup = jest.fn();

    const { getByTestId, getByPlaceholderText } = render(
      <AuthContext.Provider value={{ state, signup }}>
        <SignUp navigation={{ navigate, addListener }} />
      </AuthContext.Provider>
    );
    // when
    fireEvent.changeText(getByPlaceholderText(/username/i), username);
    fireEvent.changeText(getByPlaceholderText(/email/i), email);
    fireEvent.changeText(getByPlaceholderText(/password/i), password);
    fireEvent.press(getByTestId('signUp'));
    const flushPromises = () => new Promise(setImmediate);
    await flushPromises();
    // then
    expect(signup).toBeCalledWith({ email, password, username });
    expect(navigate).toBeCalledWith('User');
  });
});
