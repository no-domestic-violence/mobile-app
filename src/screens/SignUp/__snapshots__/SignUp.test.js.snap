// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign Up screen should match snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#cadeee",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
        Object {
          "height": "100%",
          "justifyContent": "flex-end",
        },
      ]
    }
  >
    <svg
      data-file-name="SvgLogin"
      style={
        Object {
          "position": "absolute",
          "top": 0,
        }
      }
    />
    <Text
      style={
        Object {
          "alignSelf": "flex-start",
          "fontSize": 35,
          "fontWeight": "600",
          "marginBottom": 40,
          "marginLeft": 30,
        }
      }
      testID="form-header"
    >
      Sign Up
    </Text>
    <View
      style={
        Object {
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCompleteType="off"
          autoCorrect={false}
          blurOnSubmit={false}
          editable={true}
          onChangeText={[Function]}
          placeholder="Username"
          placeholderTextColor="#6c757d"
          rejectResponderTermination={true}
          returnKeyType="next"
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#fff",
              "borderRadius": 5,
              "color": "#000",
              "flex": 1,
              "fontSize": 18,
              "fontWeight": "400",
              "minHeight": 40,
              "paddingBottom": "4%",
              "paddingLeft": "2%",
              "paddingTop": "4%",
              "width": "85%",
            }
          }
          testID="RNE__Input__text-input"
          textContentType="oneTimeCode"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <Text
        style={
          Object {
            "color": "#ff190c",
            "fontSize": 12,
            "margin": 5,
          }
        }
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "bottom": 10,
            "color": "red",
          },
        ]
      }
    >
       
    </Text>
    <View
      style={
        Object {
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCompleteType="off"
          autoCorrect={false}
          blurOnSubmit={false}
          editable={true}
          onChangeText={[Function]}
          placeholder="Email"
          placeholderTextColor="#6c757d"
          rejectResponderTermination={true}
          returnKeyType="next"
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#fff",
              "borderRadius": 5,
              "color": "#000",
              "flex": 1,
              "fontSize": 18,
              "fontWeight": "400",
              "minHeight": 40,
              "paddingBottom": "4%",
              "paddingLeft": "2%",
              "paddingTop": "4%",
              "width": "85%",
            }
          }
          testID="RNE__Input__text-input"
          textContentType="oneTimeCode"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <Text
        style={
          Object {
            "color": "#ff190c",
            "fontSize": 12,
            "margin": 5,
          }
        }
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "bottom": 10,
            "color": "red",
          },
        ]
      }
    >
       
    </Text>
    <View
      style={
        Object {
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCompleteType="off"
          autoCorrect={false}
          blurOnSubmit={false}
          editable={true}
          isInputFocused={[Function]}
          onChangeText={[Function]}
          placeholder="Password"
          placeholderTextColor="#6c757d"
          rejectResponderTermination={true}
          returnKeyType="next"
          secureTextEntry={true}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#fff",
              "borderRadius": 5,
              "color": "#000",
              "flex": 1,
              "fontSize": 18,
              "fontWeight": "400",
              "minHeight": 40,
              "paddingBottom": "4%",
              "paddingLeft": "2%",
              "paddingTop": "4%",
              "width": "85%",
            }
          }
          testID="RNE__Input__text-input"
          textContentType="oneTimeCode"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <Text
        style={
          Object {
            "color": "#ff190c",
            "fontSize": 12,
            "margin": 5,
          }
        }
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "bottom": 10,
            "color": "red",
          },
        ]
      }
    >
       
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#415889",
          "borderRadius": 20,
          "fontWeight": "600",
          "marginTop": 40,
          "opacity": 1,
          "overflow": "hidden",
        }
      }
      testID="signUp"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 14,
              "fontWeight": "600",
              "overflow": "hidden",
              "paddingBottom": 10,
              "paddingLeft": 50,
              "paddingRight": 50,
              "paddingTop": 10,
              "textAlign": "center",
            },
          ]
        }
      >
        SIGN UP
      </Text>
    </View>
    <View
      style={
        Object {
          "marginBottom": 60,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontSize": 14,
              "fontStyle": "italic",
              "marginTop": 20,
            }
          }
        >
           Have an account? Go to login
        </Text>
      </View>
    </View>
  </View>
</View>
`;
